engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -89,15 +96,13 @@ object ProgramExecutionSupport {
       }
 
     def notifyPendingCacheItems(cache: RuntimeCache): Unit = {
-      val knownKeys   = cache.getWeights.entrySet
+      val knownKeys   = cache.getWeights.keySet()
       val cachedKeys  = cache.getKeys
-      val pendingKeys = new collection.mutable.HashSet[UUID]()
+      val pendingKeys = new mutable.HashSet[UUID]()
 
-      knownKeys.forEach { e =>
-        if (e.getValue > 0) {
-          if (!cachedKeys.contains(e.getKey)) {
-            pendingKeys.add(e.getKey)
-          }
+      knownKeys.forEach { key =>
+        if (!cachedKeys.contains(key)) {
+          pendingKeys.add(key)
```

Comment:
This is the same code as before. Just it is using `keySet` instead of `entrySet`.